vcr/pe/presentation_submission.go

```diff
@@ -169,3 +170,85 @@ func (b *PresentationSubmissionBuilder) Build(format string) (PresentationSubmis
 
 	return presentationSubmission, nonEmptySignInstructions, nil
 }
+
+// Resolve returns a map where each of the input descriptors is mapped to the corresponding VerifiableCredential.
+// If an input descriptor can't be mapped to a VC, an error is returned.
+// This function is specified by https://identity.foundation/presentation-exchange/#processing-of-submission-entries
+func (s PresentationSubmission) Resolve(presentations []vc.VerifiablePresentation) (map[string]vc.VerifiableCredential, error) {
+	var envelopeJSON []byte
+	if len(presentations) == 1 {
+		// TODO: This might not be right, caller might even use a JSON array as envelope with a single VP?
+		envelopeJSON, _ = json.Marshal(presentations[0])
+	} else {
+		envelopeJSON, _ = json.Marshal(presentations)
+	}
+	var envelope interface{}
+	if err := json.Unmarshal(envelopeJSON, &envelope); err != nil {
+		return nil, fmt.Errorf("unable to convert presentations to an interface: %w", err)
+	}
+
+	result := make(map[string]vc.VerifiableCredential)
+	for _, inputDescriptor := range s.DescriptorMap {
+		resolvedCredential, err := resolveCredential(inputDescriptor.Id, 0, inputDescriptor, envelope)
+		if err != nil {
+			return nil, fmt.Errorf("unable to resolve credential for input descriptor '%s': %w", inputDescriptor.Id, err)
+		}
+		result[inputDescriptor.Id] = *resolvedCredential
+	}
+	return result, nil
+}
+
+func resolveCredential(descriptorID string, level int, mapping InputDescriptorMappingObject, value interface{}) (*vc.VerifiableCredential, error) {
```

Comment:
`descriptorID` unused

Reply:
It's passed down to the `resolveCredential` call. Oh wait...